cmake/install_helpers.cmake

```diff
@@ -126,9 +132,7 @@ function(ginkgo_install)
         COMPONENT Ginkgo_Development)
 
     # Export package for use from the build tree
-    if (GINKGO_EXPORT_BUILD_DIR)
-        export(PACKAGE Ginkgo)
-    endif()
+    export(PACKAGE Ginkgo)
```

Comment:
could you explain more on this?
It does nothing without CMAKE_EXPORT_PACKAGE_REGISTRY=ON. Thus, it does not write anything to the user registry like ~/.cmake in the linux. (it does not affect us to use build directory as the external library)
never noticed cmake contains a user package registry before

Reply:
I think that already covers it. Without `CMAKE_EXPORT_PACKAGE_REGISTRY=ON` this is a no-op, so we don't need to guard it. Also setting `GINKGO_EXPORT_BUILD_DIR` had no effect on the other CMake option, so no need for that.
The CMake registry basically just stores the paths to the packages.